docker-action/entrypoint.sh

```diff
@@ -1,31 +1,73 @@
 #!/bin/bash
 
+DEBUG_MODE=1
+
+if [[ $DEBUG_MODE -eq 1 ]]; then
+    DEBUG_OUT="/dev/stdout"
+else
+    DEBUG_OUT="/dev/nil"
+fi
+
 RED='\033[0;31m'
 GREEN='\033[0;32m'
 YELLOW='\033[1;33m'
 RESET='\033[0m' # No Color
 
+echo "[DEBUG] Github Workspace: $GITHUB_WORKSPACE" > $DEBUG_OUT
 REPOSITORY_NAME=$(echo "$GITHUB_REPOSITORY" | awk -F / '{print $2}' | sed -e "s/:refs//")
 
+# returns the absolute path from a base path ($1) and a list of paths relative
+# to the base path (${@:2}). Also works if one of the argument paths is an
+# absolute path. Note: does not handle paths that contain a space.
+getFileListInDir () (
+    local LOCATION=$1
+    cd -- "$LOCATION"
+    # the tail of the list of arguments (i.e., the args without
+    # the function name and the first argument (LOCATION) are
+    # the list of paths to be processed.
+    echo "$(echo "${@:2}" | xargs realpath | tr '\n' ' ')"
+)
+
 if [[ $INPUT_PROJECTLOCATION ]]; then
     PROJECT_LOCATION="$GITHUB_WORKSPACE/$INPUT_PROJECTLOCATION"
 else
     PROJECT_LOCATION="$GITHUB_WORKSPACE/$REPOSITORY_NAME"
 fi
 
-if [[ $INPUT_SRCDIRECTORY ]]; then
-    GOBRAINPUT_LOCATION="$PROJECT_LOCATION/$INPUT_SRCDIRECTORY"
-else
-    GOBRAINPUT_LOCATION="$PROJECT_LOCATION"
+GOBRA_JAR="/gobra/gobra.jar"
+JAVA_ARGS="-Xss$INPUT_JAVAXSS -Xmx$INPUT_JAVAXMX -jar $GOBRA_JAR"
+GOBRA_ARGS="--backend $INPUT_VIPERBACKEND --chop $INPUT_CHOP"
+
+if [[ $INPUT_RECURSIVE -eq 1 ]]; then
+    GOBRA_ARGS="--recursive $GOBRA_ARGS"
 fi
 
-GOBRA_JAR="/gobra/gobra.jar"
+if [[ $INPUT_FILES ]]; then
+    RESOLVED_PATHS="$(getFileListInDir $PROJECT_LOCATION $INPUT_FILES)"
+    echo "[DEBUG] Project Location: $PROJECT_LOCATION" > $DEBUG_OUT
+    echo "[DEBUG] Input Files: $INPUT_FILES" > $DEBUG_OUT
+    echo "[DEBUG] Resolved Paths: $RESOLVED_PATHS" > $DEBUG_OUT
+    GOBRA_ARGS="-i $RESOLVED_PATHS $GOBRA_ARGS"
+fi
 
-JAVA_ARGS="-Xss$INPUT_JAVAXSS -Xmx$INPUT_JAVAXMX -jar $GOBRA_JAR"
-GOBRA_ARGS="-i $GOBRAINPUT_LOCATION -r --backend $INPUT_VIPERBACKEND --chop $INPUT_CHOP"
+if [[ $INPUT_PACKAGES ]]; then
+    if [[ $INPUT_RECURSIVE -eq 1 ]]; then
+        # If in recursive mode, INPUT_PACKAGES are package names
```

Comment:
Also, `-p` together with `--recursive` is imho invalid

Reply:
Sorry, my mistake. At any rate, I think I will drop the --includePackages option from the CI, as it can be easily replaced by the `-p` option. This also prevents confusion from users from seeing options with a similar description. Do you oppose to this?